src/ImageSharp/Memory/Allocators/MemoryAllocator.cs

```diff
@@ -20,6 +24,12 @@ public abstract class MemoryAllocator
     /// </summary>
     public static MemoryAllocator Default { get; } = Create();
 
+    internal long MemoryGroupAllocationLimitBytes { get; private set; } = Environment.Is64BitProcess ?
+        4L * OneGigabyte :
+        OneGigabyte;
+
+    internal int SingleBufferAllocationLimitBytes { get; private set; } = OneGigabyte;
```

Comment:
Actually, there is no good reason to set this to 1GB. Contiguous buffers have a natural limit of `int.MaxValue ~ 2GB` so I think we should set the contiguous limit to `min(int.MaxValue, MemoryAllocatorOptions.AllocationLimitMegabytes * 1MB)`. That would get rid of all obscure logic and make the semantics of `MemoryAllocatorOptions.AllocationLimitMegabytes` very obvious.

Reply:
Yeah. I like that. The only risk we have with this stuff is in our decoders which can be easily mitigated with some care (PNG tweaks)